Invalid manifest format after upgrading from 0.3 to 0.5

I just upgraded our project from plugin version 0.3 to 0.5 to benefit from the `.tar` support.
After upgrading, some `.war` files fail to start, the exception is:

```
SEVERE: ContainerBase.addChild: start: 
org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Catalina].StandardHost[localhost].StandardContext[/app-ws]]
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:153)
        at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:725)
        at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:701)
        at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:717)
        at org.apache.catalina.startup.HostConfig.deployWAR(HostConfig.java:939)
        at org.apache.catalina.startup.HostConfig$DeployWar.run(HostConfig.java:1812)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.apache.catalina.LifecycleException: Failed to start component [org.apache.catalina.webresources.StandardRoot@6ed89845]
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:153)
        at org.apache.catalina.core.StandardContext.resourcesStart(StandardContext.java:4928)
        at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5058)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:147)
        ... 10 more
Caused by: org.apache.catalina.LifecycleException: Failed to initialize component [org.apache.catalina.webresources.JarResourceSet@71310103]
        at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:107)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:136)
        at org.apache.catalina.webresources.StandardRoot.startInternal(StandardRoot.java:699)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:147)
        ... 13 more
Caused by: java.lang.IllegalArgumentException: java.io.IOException: invalid manifest format
        at org.apache.catalina.webresources.JarResourceSet.initInternal(JarResourceSet.java:96)
        at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:102)
        ... 16 more
Caused by: java.io.IOException: invalid manifest format
        at java.util.jar.Manifest.read(Manifest.java:225)
        at java.util.jar.Manifest.<init>(Manifest.java:69)
        at java.util.jar.JarFile.getManifestFromReference(JarFile.java:194)
        at java.util.jar.JarFile.getManifest(JarFile.java:180)
        at org.apache.catalina.webresources.JarResourceSet.initInternal(JarResourceSet.java:94)
        ... 17 more
```

unfortunately the logs are not giving much details :-(

I noticed this [commit](https://github.com/Zlika/reproducible-build-maven-plugin/commit/664d0593365ada02ade449447c37c0400dd5f2b5), which could be probable cause for the above error, but I can't see anything wrong with it.

Also I tried inspecting the `MANIFEST.MF` file from the war META-INF dir, but that specific file does not appear to be the problem, so it must be somewhere else.

--- a/src/main/java/io/github/zlika/reproducible/SortManifestFileStripper.java
+++ b/src/main/java/io/github/zlika/reproducible/SortManifestFileStripper.java
@@ -61,12 +61,12 @@ final class SortManifestFileStripper implements Stripper
             }
         }
     };
-
+    
     @Override
     public void strip(File in, File out) throws IOException
     {
         final List<String> lines = Files.readAllLines(in.toPath(), StandardCharsets.UTF_8);
-
+        
         try (final BufferedWriter writer = new BufferedWriter(
                 new OutputStreamWriter(new FileOutputStream(out), StandardCharsets.UTF_8)))
         {
@@ -81,7 +81,7 @@ final class SortManifestFileStripper implements Stripper
             }
         }
     }
-
+    
     private List<String> sortManifestSections(List<String> lines)
     {
         final List<List<String>> sections = new ArrayList<>();
@@ -106,14 +106,14 @@ final class SortManifestFileStripper implements Stripper
         {
             sections.add(currentSection);
         }
-
+        
         return sections.stream()
                         .map(list -> sortAttributes(list))
                         .map(list -> String.join("", list))
                         .sorted(MANIFEST_ENTRY_COMPARATOR)
                         .collect(Collectors.toList());
     }
-
+    
     private List<String> sortAttributes(List<String> lines)
     {
         final List<String> attributes = new ArrayList<>();

--- a/src/main/java/io/github/zlika/reproducible/SortManifestFileStripper.java
+++ b/src/main/java/io/github/zlika/reproducible/SortManifestFileStripper.java
@@ -51,7 +51,7 @@ final class SortManifestFileStripper implements Stripper
             {
                 return -1;
             }
-            else if (o2.startsWith("Name:") && !o2.startsWith("Name:"))
+            else if (o2.startsWith("Name:") && !o1.startsWith("Name:"))
             {
                 return 1;
             }
@@ -61,12 +61,12 @@ final class SortManifestFileStripper implements Stripper
             }
         }
     };
-    
+
     @Override
     public void strip(File in, File out) throws IOException
     {
         final List<String> lines = Files.readAllLines(in.toPath(), StandardCharsets.UTF_8);
-        
+
         try (final BufferedWriter writer = new BufferedWriter(
                 new OutputStreamWriter(new FileOutputStream(out), StandardCharsets.UTF_8)))
         {
@@ -81,7 +81,7 @@ final class SortManifestFileStripper implements Stripper
             }
         }
     }
-    
+
     private List<String> sortManifestSections(List<String> lines)
     {
         final List<List<String>> sections = new ArrayList<>();
@@ -106,14 +106,14 @@ final class SortManifestFileStripper implements Stripper
         {
             sections.add(currentSection);
         }
-        
+
         return sections.stream()
                         .map(list -> sortAttributes(list))
                         .map(list -> String.join("", list))
                         .sorted(MANIFEST_ENTRY_COMPARATOR)
                         .collect(Collectors.toList());
     }
-    
+
     private List<String> sortAttributes(List<String> lines)
     {
         final List<String> attributes = new ArrayList<>();

--- a/src/test/java/io/github/zlika/reproducible/ManifestStripperTest.java
+++ b/src/test/java/io/github/zlika/reproducible/ManifestStripperTest.java
@@ -35,9 +35,9 @@ public class ManifestStripperTest
     {
         final File out = File.createTempFile("manifest", null);
         out.deleteOnExit();
-	
+        
         new ManifestStripper().strip(new File(this.getClass().getResource("MANIFEST.MF").getFile()), out);
-	
+        
         final byte[] expected = Files.readAllBytes(new File(
                                     this.getClass().getResource("MANIFEST-stripped.MF").getFile()).toPath());
         final byte[] actual = Files.readAllBytes(out.toPath());

--- a/src/test/java/io/github/zlika/reproducible/ManifestStripperTest.java
+++ b/src/test/java/io/github/zlika/reproducible/ManifestStripperTest.java
@@ -28,20 +28,21 @@ public class ManifestStripperTest
 {
     /**
      * Tests stripping on a reference Manifest file.
-     * @throws IOException 
+     * @throws IOException
      */
     @Test
     public void testStripManifest() throws IOException
     {
         final File out = File.createTempFile("manifest", null);
-        out.deleteOnExit();
-        
+        //out.deleteOnExit();
+
         new ManifestStripper().strip(new File(this.getClass().getResource("MANIFEST.MF").getFile()), out);
-        
+
         final byte[] expected = Files.readAllBytes(new File(
                                     this.getClass().getResource("MANIFEST-stripped.MF").getFile()).toPath());
         final byte[] actual = Files.readAllBytes(out.toPath());
+        System.out.println("-"+new String(actual)+"-");
         Assert.assertArrayEquals(expected, actual);
-        out.delete();
+        //out.delete();
     }
 }

--- a/src/test/java/io/github/zlika/reproducible/ManifestStripperTest.java
+++ b/src/test/java/io/github/zlika/reproducible/ManifestStripperTest.java
@@ -34,15 +34,14 @@ public class ManifestStripperTest
     public void testStripManifest() throws IOException
     {
         final File out = File.createTempFile("manifest", null);
-        //out.deleteOnExit();
+        out.deleteOnExit();
 
         new ManifestStripper().strip(new File(this.getClass().getResource("MANIFEST.MF").getFile()), out);
 
         final byte[] expected = Files.readAllBytes(new File(
                                     this.getClass().getResource("MANIFEST-stripped.MF").getFile()).toPath());
         final byte[] actual = Files.readAllBytes(out.toPath());
-        System.out.println("-"+new String(actual)+"-");
         Assert.assertArrayEquals(expected, actual);
-        //out.delete();
+        out.delete();
     }
 }

--- a/src/test/java/io/github/zlika/reproducible/ManifestStripperTest.java
+++ b/src/test/java/io/github/zlika/reproducible/ManifestStripperTest.java
@@ -28,7 +28,7 @@ public class ManifestStripperTest
 {
     /**
      * Tests stripping on a reference Manifest file.
-     * @throws IOException
+     * @throws IOException On error.
      */
     @Test
     public void testStripManifest() throws IOException

--- a/src/test/java/io/github/zlika/reproducible/ManifestStripperTest.java
+++ b/src/test/java/io/github/zlika/reproducible/ManifestStripperTest.java
@@ -35,9 +35,9 @@ public class ManifestStripperTest
     {
         final File out = File.createTempFile("manifest", null);
         out.deleteOnExit();
-
+	
         new ManifestStripper().strip(new File(this.getClass().getResource("MANIFEST.MF").getFile()), out);
-
+	
         final byte[] expected = Files.readAllBytes(new File(
                                     this.getClass().getResource("MANIFEST-stripped.MF").getFile()).toPath());
         final byte[] actual = Files.readAllBytes(out.toPath());

--- a/src/test/resources/io/github/zlika/reproducible/MANIFEST-stripped.MF
+++ b/src/test/resources/io/github/zlika/reproducible/MANIFEST-stripped.MF
@@ -1,6 +1,6 @@
 Manifest-Version: 1.0
 Archiver-Version: Plexus Archiver
-Description: This is a dummy description with 2 trailing white spaces 
+Description: This is a dummy description with 2 trailing white spaces
   
 Import-Package: org.eclipse.core.runtime;version="[3.4,4)";resolution:
  =optional,org.apache.commons.logging;resolution:=optional;version="[1
@@ -8,6 +8,12 @@ Import-Package: org.eclipse.core.runtime;version="[3.4,4)";resolution:
  n:=optional,org.eclipse.osgi.service.datalocation;version="[1.3,2)";r
  esolution:=optional,org.osgi.framework;version="[1.7,2)";resolution:=
  optional
+
+Name: com/acme/fonts/
+Implementation-Title: com.acme.fonts
+Implementation-Vendor: Acme, Inc
+Specification-Title: A font extension
+Specification-Vendor: ACME, Inc.
 
 Name: foo/bar/a
 Sealed: false

--- a/src/test/resources/io/github/zlika/reproducible/MANIFEST-stripped.MF
+++ b/src/test/resources/io/github/zlika/reproducible/MANIFEST-stripped.MF
@@ -1,6 +1,6 @@
 Manifest-Version: 1.0
 Archiver-Version: Plexus Archiver
-Description: This is a dummy description with 2 trailing white spaces
+Description: This is a dummy description with 2 trailing white spaces 
   
 Import-Package: org.eclipse.core.runtime;version="[3.4,4)";resolution:
  =optional,org.apache.commons.logging;resolution:=optional;version="[1

--- a/src/test/resources/io/github/zlika/reproducible/MANIFEST.MF
+++ b/src/test/resources/io/github/zlika/reproducible/MANIFEST.MF
@@ -9,7 +9,7 @@ Import-Package: org.eclipse.core.runtime;version="[3.4,4)";resolution:
  n:=optional,org.eclipse.osgi.service.datalocation;version="[1.3,2)";r
  esolution:=optional,org.osgi.framework;version="[1.7,2)";resolution:=
  optional
-Description: This is a dummy description with 2 trailing white spaces
+Description: This is a dummy description with 2 trailing white spaces 
   
 
 Name: foo/bar/b

--- a/src/test/resources/io/github/zlika/reproducible/MANIFEST.MF
+++ b/src/test/resources/io/github/zlika/reproducible/MANIFEST.MF
@@ -9,7 +9,7 @@ Import-Package: org.eclipse.core.runtime;version="[3.4,4)";resolution:
  n:=optional,org.eclipse.osgi.service.datalocation;version="[1.3,2)";r
  esolution:=optional,org.osgi.framework;version="[1.7,2)";resolution:=
  optional
-Description: This is a dummy description with 2 trailing white spaces 
+Description: This is a dummy description with 2 trailing white spaces
   
 
 Name: foo/bar/b
@@ -20,3 +20,9 @@ Sealed: false
 
 Name: foo/bar/baz/
 Sealed: true
+
+Name: com/acme/fonts/
+Specification-Title: A font extension
+Specification-Vendor: ACME, Inc.
+Implementation-Title: com.acme.fonts
+Implementation-Vendor: Acme, Inc

